src/node/txdownloadman_impl.cpp

```diff
@@ -357,8 +357,6 @@ node::RejectedTxTodo TxDownloadManagerImpl::MempoolRejectedTx(const CTransaction
 {
     const CTransaction& tx{*ptx};
     // Results returned to caller
-    // Whether we should call AddToCompactExtraTransactions at the end
-    bool add_extra_compact_tx{first_time_failure};
```

Comment:
Any thoughts on storing transactions from outbounds only? In this function we'd be able to query who the announcers were (that we still remember). net_processing also knows who sent the transaction.